%!TEX root = ../chapter2.tex
%******************************
%	 Discussion 
%*****************************

\section{Discussion}

This chapter addressed a statistical problem that obstructed the analysis of the previous chapter, where changes in expression variability were confounded by changes in mean expression. 
Here, this problem is resolved by introducing a hierarchical Bayesian formulation that extends the BASiCS framework. 
In this formulation, an additional set of residual over-dispersion parameters $\epsilon_i$ that are not confounded by changes in mean expression are introduced. 
This extension ensures a broader applicability of the BASiCS software and allows statistical testing of changes in variability that are not confounded by mean expression.  \\ 

\cor{The original implementation of BASiCS for testing changes in mean expression and expression variability aimed at borrowing information across all cells, all genes and both conditions.
However, as cell-specific scaling factors and gene-specific mean expression parameters are jointly inferred by BASiCS, this led to an unidentifibility problem: a global offset (associated to global differences in mRNA content) between the two conditions is unknown a priori. 
This issue did not arise in most differential expression methods for bulk RNA-seq data (such as DESeq2 \cite{Love2014} and edgeR \cite{Robinson2009}), because they typically adopted a sequential approach: they first estimate cell-specific scaling factors that are subsequently used as fixed offsets in the downstream model for differential expression. 
Instead, BASiCS resolves the unidentifiability issue by performing inference on both conditions independently, performing a post-hoc offset correction.\\
This approach raises the question if parameter estimates are comparable between both conditions or if differences are cause due to over-fitting.
We therefore validated the robustness of the model in multiple ways:\\
1. We ran the model on down-sampled cell populations and found that (i) the majority of model parameters show similar estimates independent of the sample size and expression levels, and (ii) the regression stabilises variability estimates for lowly expressed genes. \\
2. We simulated data and showed that (i) the false positive rate is less than 5\% for mean and variability estimates and that (ii) the true positive rate reaches 100\% for large sample size. \\
3. we compared BASiCS parameter estimates to absolute transcript counts using smFISH and observed a near perfect estimation of mean expression but lower correlation for variability and residual variability measures.\\
We are therefore confident that BASiCS correctly estimates mean expression and expression variability per condition and that this allows comparison of parameter estimates between conditions.}\\

Other, dataset-specific factors that affect the robustness of posterior parameter estimation include sequencing depth, number of cells and heterogeneity within the population. 
BASiCS has been designed to be run on homogeneous populations of cells to avoid variation introduced by correlated sub-structure within the data. 
Furthermore, the use of UMIs for quantifying transcript counts can also improve variability estimation \citep{Grun2014}. \\

Our method enables the characterisation of the extent and nature of variable gene expression in CD4\plus{} T cell activation and differentiation. 
Firstly, we observe that during acute activation of naive T cells, genes of the biosynthetic machinery are homogeneously up-regulated, while specific immune-related genes are more heterogeneously up-regulated. In particular, increased variability in expression of the apoptosis-inducing Fas ligand \citep{Strasser2009} and the inhibitory ligand PD-L1 \citep{Chikuma2016} suggests a mechanism by which newly activated cells might suppress re-activation of effector cells, thereby dynamically modulating the population response to activation. 
Likewise, more variable expression of \emph{Smad3}, which translates inhibitory \gls{Tgf}$\beta$ signals into transcriptional changes \citep{Delisle2013}, may indicate increased diversity in cellular responses to this signal. 
Increased variability in \textit{Pou2f2} (Oct2) expression after activation suggests heterogeneous activities of the NF-$\kappa$B and/or \gls{NFAT} signalling cascades that control its expression \citep{Mueller2013}.
Moreover, we detect up-regulated and more variable \textit{Il2} expression, suggesting heterogeneous Il2 protein expression, which is known to enable T cell population responses \citep{Fuhrmann2016}. 

Additionally, we studied changes in gene expression variability during CD4\plus{} T cell differentiation towards a Th1 and Tfh cell state over a 7 day time course after \textit{in vivo} malaria infection \citep{Lonnberg2017}. 
Our analysis provides several insights into this differentiation system. 
Firstly, we observe a tighter regulation in gene expression among genes that do not change in mean expression during differentiation at day 4. 
At this point, divergence of Th1 and Tfh differentiation was previously identified \citep{Lonnberg2017}. 
The decrease in variability on day 4 is potentially due to induction of a strong pan-lineage proliferation programme. 
However, we observe that not all genes follow this trend and uncover four different patterns of variability changes. 
Secondly, we observe that several Tfh and Th1 lineage-associated genes change in expression variability between days 2 and 4. For example, we noted a decrease in variability for one key Th1 regulator, \textit{Tbx21} (encoding Tbet), which suggests that a subset of cells may have already committed to the Th1 lineage at day 2. 
Three additional Th1 lineage-associated genes also followed this trend (\textit{Ahnak}, \textit{Ctsd}, \textit{Tmem154}). 
These data suggest that differentiation fate decisions may arise as early as day 2 in subpopulations within this system, resulting in high gene expression variability. 
Such an effect is in accordance with the early commitment to effector T cell fates that was previously observed during viral infection \citep{Choi2011}. 
As these results illustrate, diversity in the differentiation state within a population of T cells can drive our differential variability results. 
To further dissect these results, subsequent analyses such as the pseudo-time inference used in \cite{Lonnberg2017} could be used to characterise a continuous differentiation process.\\

Finally, I confirmed that the regression BASiCS model can be used to study heterogeneous differentiation processes in droplet-based scRNA-Seq data. 
While the sparsity of these data can lead to unstable posterior inference, the use of UMIs allows for robust measurement of transcript counts. 
Therefore, posterior estimates of model parameters can be used to detect changes in mean expression and in transcriptional variability. 
In the case of somitogenesis, I validated known Wnt and Fgf signalling pathways in pre-somitic mesoderm cells and newly discovered a possible priming for somitic mesoderm cells to later form limb muscles. 
Identifying these genes supports a deeper understanding of cell fate decisions in the developing embryo.\\

In sum, our model provides a robust tool for understanding the role of heterogeneity in gene expression during cell fate decisions. 
This tool can be widely applied to systems where strong expression changes occur and for sparse droplet-based scRNA-Seq datasets. 
It allows the statistical assessment of changes in variability due to available uncertainty estimates. 
A drawback of this framework is the restriction to relatively few (up to 1000) cells that can be modelled per run. 
In \textbf{Chapter 5}, I will discuss current and future extensions to the Bayesian inference of generative models that allows the estimation of model parameters using more than 1 million cells.


