%!TEX root = ../chapter2.tex
%******************************
%	 Introduction 
%*****************************

\section{Introduction}

\todo{EXPAND INTRODUCTION} 

Heterogeneity in gene expression within a population of single cells can arise from a variety of factors. Structural differences in gene expression within a cell population can reflect the presence of sub-populations of functionally different cell types \citep{Zeisel2015, Paul2015}. Alternatively, in a seemingly homogeneous population of cells, so-called unstructured expression heterogeneity can be linked to intrinsic or extrinsic noise \citep{Elowitz2002}. 
Changes in physiological cell states (such as cell cycle, metabolism, abundance of transcriptional/translational machinery and growth rate) represent extrinsic noise, which has been found to influence expression variability within cell populations \citep{Keren2015, Buettner2015, Zeng2017}. Intrinsic noise can be linked to epigenetic diversity \citep{Smallwood2014}, chromatin rearrangements \citep{Buenrostro2015}, as well as the genomic content of single genes, such as the presence of TATA-box motifs and the abundance of nucleosomes around the transcriptional start site \citep{Hornung2012}.  \\ 

Single-cell RNA sequencing (scRNAseq) generates transcriptional profiles of single cells, allowing the study of cell-to-cell heterogeneity on a transcriptome-wide \citep{Grun2014} and single gene level \citep{Goolam2016}. Consequently, this technique can be used to study unstructured cell-to-cell variation in gene expression within and between homogeneous cell populations (i.e.~where no distinct cell sub-types are present). Increasing evidence suggests that this heterogeneity plays an important role in normal development \citep{Chang2008} and that control of expression noise is important for tissue function \citep{BaharHalpern2015}. For instance, molecular noise was shown to increase before cells commit to lineages during differentiation \citep{Mojtahedi2016}, while the opposite is observed once an irreversible cell state is reached \citep{Richard2016}. A similar pattern occurs 
during gastrulation, where expression noise is high in the uncommitted inner cell mass compared to the committed epiblast and where an increase in heterogeneity is observed when cells exit the pluripotent state and form the uncommitted epiblast \citep{Mohammed2017}. \\

Motivated by scRNAseq, recent studies have extended traditional differential expression analyses to explore more general patterns that characterise differences between cell populations \citep[e.g.~][]{Korthauer2016}. In particular, BASiCS \citep{Vallejos2015,Vallejos2016a} introduced a probabilistic tool to assess differences in cell-to-cell heterogeneity between two or more cell populations. This feature has led to, for example, insights into the context of immune activation and ageing \citep{Martinez-jimenez2017}. To meaningfully assess changes in biological variability across the entire transcriptome, two main confounding effects must be taken into account: differences due to artefactual technical noise and differential variability between populations that is driven by changes in mean expression. The latter arises because biological noise is negatively correlated with protein abundance \citep{Bar-Even2006, Newman2006, Taniguchi2011} or mean RNA expression \citep{Brennecke2013, Antolovic2017}. To address these two confounding effects, BASiCS separates biological noise from technical variability  by borrowing information from synthetic RNA spike-in molecules. Additionally to acknowledge the variance-mean relationship, it restricts differential variability testing to those genes with equal mean expression across populations. \\

This article extends the statistical model implemented in BASiCS by implementing a more general approach to account for the aforementioned confounding effects. Firstly, we derive a residual measure of cell-to-cell transcriptional variability that is not confounded by mean expression. This is used to define a probabilistic rule to robustly highlight changes in variability, even for differentially expressed genes. Unlike previous related methods \citep[e.g.~][]{Kolodziejczyk2015cell}, our approach directly performs gene-specific statistical testing between two conditions using a readily available measure of uncertainty. Secondly, by exploiting concepts from measurement error models, our method is extended to address experimental designs where spike-in sequences are not available. This is particularly critical due to the increasing popularity of droplet-based technologies. \\

Using our approach, we identify a synchronisation of  biosynthetic machinery components in CD4$^+$ T cells upon early immune activation as well as an increased variability in the expression of genes related to CD4$^+$ T cell immunological function.
Furthermore, we detect evidence of early cell fate commitment of CD4$^+$ T cells during malaria infection characterized by a decrease in \textit{Tbx21} expression heterogeneity and a rapid collapse of global transcriptional variability after infection. These results highlight biological insights into T cell activation and differentiation that are only revealed by jointly studying changes in mean expression and variability.
