%!TEX root = ../intro.tex
%******************************
%	 Other applications of scRNAseq
%*****************************

\section{General applications of scRNA-Seq in biology}

The following section outlines the broad spectrum of research fields that benefit from the development of scRNA-Seq technologies. Whole transcriptomic read-outs of individual cells allowed the in-depth characterisation of embryonic development, hematopoiesis, immune responses, allowed the detection of rare cell types and lead to new insights into disease progression including cancer development. 

\subsection{Atlas-type approaches}

Until recently, scRNA-Seq technologies were used to generate transcriptomes of less than a thousand cells to address specific questions in cellular systems such as cell-type heterogeneity, allele-specific expression or pseudo-temporal trajectories in gene expression \citep{Kolodziejczyk2015review}. With the development of scRNA-Seq technologies that massively increased the throughput of cell capture and data generation, cellular composition of whole tissues and organisms can be assayed. The largest of these so called "atlases" to date is the 10X Genomics\textsuperscript{\textregistered}{} brain dataset comprising 1.3 million cells from embryonic mice. I was generated using 133 libraries sequenced on 11 Illumina HiSeq\textsuperscript{\textregistered}{} 4000 flowcells \citep{Note2017}. This experiment has been performed to exemplify the applicability of the commercial 10X genomics platform to generate more than 10 billion transcriptomes of individual cells across the human body as envisioned by the Human Cell Atlas Consortium \citep{Regev2017}.\\

So far, examples of transcriptional atlases that comprise hundreds of thousands of cells are the mouse cell atlas, a thymus organogenesis atlas, an ageing lung atlas and the full characterisation of cell-types in \textit{C. elegans}. Similar to CytoSeq, Microwell-Seq was developed to capture more than 400,000 cells covering all mouse organ. This analysis reveals rare cell types, for example 2-cell-stage like mouse embryonic stem cells and allows the construction of a cross-tissue correlation network \cite{Han2018}. Similarly, the \emph{Tabula Muris} aimed at detecting all major cell type across 20 organs of the mouse. Here, the Tabular Muris Consortium decided to used droplet-based 3'-end scRNA-Seq and FACS-based full length transcript analysis to generate (i) a broad atlas and (ii) an in-depth characterisation of each tissue \citep{Quake2018}. Cao \emph{et al.}, 2017 generated more than 40,000 cells from the L2 stage \emph{C. elegans} using sci-RNA-Seq and identified nineteen distinct cell types and seven mixed cell types. Furthermore, this atlas allows the dissection of neuronal cell types that split across seven clusters \citep{Cao2017}. To study thymus development, Kernfeld \emph{et al.}, 2018 generated around 25,000 transcriptomes of individual cells from the embryonic thymus at E12.5, E13.5, E14.5, E15.5, E16.5, E17.5, E18.5, and P0. This experimental set-up resolves the temporal development of immune cell types such as T cells, myeolid cells, natural killer cells, innate lymphoid cells, and $\gamma{}\delta{}$ T cells as well as thymic epithelial cells \citep{Kernfeld2018}. Finally, to study the effect of ageing on a whole tissue, Angelidis \emph{et al.}, 2018 isolated 14,000 cells from lungs of young and old animals and found (i) and increase in transcriptional noise during ageing and (ii) altered transcriptional profiles of alveolar macrophages and type 2 pneumocytes \citep{Angelidis2018}.\\

The following paragraphs summarise scRNA-Seq applications which aimed at more targeted analysis of regulatory processes.

\subsection{Developmental biology}

For years, the development of new scRNA-Seq technologies and algorithms to perform data analysis uncovered driving factors in development and cell fate decisions \citep{Griffiths2018}. An early study during early mouse embryonic development identified that transcriptional differences between the two cells in the 2-cell stage embryo increase from the zygote to late 2-cell stage embryos. This is caused by an initial partitioning error where transcripts are unevenly distributed between the daughter cells and later on elevated by the onset of transcription coupled to transcriptional noise \citep{Piras2014, Shi2015a}. A reproducible distribution of transcripts in the first cell division was also detected by Biase \emph{et al.}, 2014 \cite{Biase2014}. These biases between cells at the 2-cell stage propagate to form transcription biases at the 4-cell stage to for the pluripotent inner cell mass or the extra-embryonic trophoectoderm \citep{Goolam2016, Shi2015a}. To obtain a more complete view on gastrulation in the mouse, Scialdone \emph{et al.} captured cells from the epiblast at E6.5 and mesodermal cells at E7.0, E7.5 and E7.75. The authors also sampled cells from \emph{Tal1} knock-out animals and showed that this transcription factor is the driving regulator for blood develeopment \citep{Scialdone2016}. \\

This year, large-scale scRNA-Seq studies profiled organogenesis in the mouse and zebrafish. Ibarra-Soria \emph{et al.} sampled more than 20,000 cells from E8.5 embryos following gastrulation and identified 20 major cell-types including different mesoderm lineages, neural progenitor cells, blood, gut and extra-embryonic cells. They further used this data to dissect gut formation and to find oscillating expression patterns during somitogenesis \citep{Ibarra-Soria2018}. Similarly, inDrop and Drop-Seq approaches were used to generate $\sim$7000 cells from \gls{Dmelanogaster} embryos at the onset of gastrulation or to generate more than 90,000 cells from the zebrafish embryo during the first day of development \citep{Wagner2018}.



Embryonic development

Lineage tracing: McKenna, Raj, Alemany, Frieda, Wagner

Spatial transcriptomics: Satija, Achim, Merfish, Seqfish, and DistMap [Karaiskos], tomoseq

\todo{Gastrulation and early mouse development and what Tang is doing}
scGESTALT, MEMOIR, van Oudenaarden lineage tracing

\subsection{Cell-type evolution} 

\todo{describe recent papers from Heather}

\subsection{Immunology}

\todo{Sarah's work and Weizmann}

\subsection{Tissue function}

\todo{Mammary gland, liver}

\subsection{Cancer}

\todo{Avivs papers}