%!TEX root = ../intro.tex
%******************************
%	 Outline 
%*****************************


\section{Outline}

The overarching topic of this thesis is the quantification and interpretation of transcriptional noise as measured by single-cell RNA sequencing. \textbf{Chapter 1} presents an initial experiment to study how  transcriptional noise effects the immune system. We see that transcriptional noise increases across multiple immune response genes during ageing which therefore could explain a disrupted immune response in older individuals. This finding has been published in following paper:\\

Celia P. Martinez-Jimenez$^\ast$, Nils  Eling$^\ast$, Hung-Chang Chen, Catalina A. Vallejos, Aleksandra Kolodziejczyk, Frances Connor, Lovorka Stojic, Tim F. Rayner, Michael J. T. Stubbington, Sarah A. Teichmann, Maike de la Roche, John C. Marioni, Duncan T. Odom. Ageing increases cell-to-cell transcriptional variability upon immune stimulation. \emph{Science}, 1436: 1433-1436, 2017, ($^\ast$ equal contributions) \\

When studying changes in variability between two conditions, the analysis for this project was restricted to genes that did not change in mean expression due to a strong confounding between variability and mean expression. In \textbf{chapter 2}, I therefore extended the statistical framework that I used for the analysis to correct for this confounding effect. This correction lead to (i) a stabilization of model parameters, (ii) expansion of the gene set that can be tested for changes in variability and (iii) a novel way of interpreting transcriptional dynamics. This project has been published as:\\

Nils Eling, Arianne C. Richard, Sylvia Richardson, John C. Marioni, Catalina A. Vallejos. Robust expression variability testing reveals heterogeneous T cell responses. \emph{Cell Systems}, In press, 2018 \\

The extended model offers the unique opportunity to study changes in variability across multiple cell types even when mean expression changes. In \textbf{chapter 3}, I apply the newly developed model to test changes in variability over pseudotime. This data has been generated to dissect the transcriptional dynamics during mouse spermatogenesis and has been submitted under: \\

Christina Ernst$^\ast$, Nils Eling$^\ast$, Celia P. Martinez-Jimenez, John C. Marioni, Duncan T. Odom. Staged developmental mapping and X chromosome transcriptional dynamics during mouse spermatogenesis. \emph{bioRxiv}, 2018, ($^\ast$ equal contributions)


\section{Other contributions}

Contributions to papers that are not discussed in this thesis are as follows:

\begin{itemize}
\item Kaia Achim$^\ast$, Nils Eling$^\ast$, Hernando Martinez Vergara, Paola Yanina Bertucci, Jacob Musser, Pavel Vopalensky, Thibaut Brunet, Paul Collier, Vladimir Benes, John C. Marioni, Detlev Arendt. Whole-Body Single-Cell Sequencing Reveals Transcriptional Domains in the Annelid Larval Body. \emph{Molecular Biology and Evolution}, 35: 1047-1062, 2018, ($^\ast$ equal contributions)\\
\item Christina Ernst, Jeremy Pike, Sarah J. Aitken, Hannah K. Long, Nils Eling, Lovorka Stojic, Michelle C. Ward, Frances Connor, Timothy F. Rayner, Margus Lukk, Robert J. Klose, Claudia Kutter, Duncan T Odom. Successful transmission and transcriptional deployment of a human chromosome via mouse male meiosis. \emph{eLife}, 5: e20235, 2016 
\end{itemize}