%!TEX root = ../main.tex
%******************************
%	 Discussion 
%*****************************

\chapter{Conclusion and future directions}  

\vspace{-5mm}

My work focused on the statistical quantification and dissection of transcriptional noise in biological systems such as the activation response of CD4\plus{} T cells. Firstly, in collaboration with Celia P. Martinez-Jimenez, I used scRNA-Seq data of CD4\plus{} T cells to identify an age-related increase in transcriptional noise within a set of immune response genes (see \textbf{Chapter 2}).   
While technological and computational advances of the recent years facilitate the quantification of biological noise across a range of cell types and tissues, major challenges remain regarding robust measurement, mathematical modelling and experimental validation. 

\section{Biological role of noise}

So far, quantification of expression noise on a genome wide scale is only possible by single-cell RNA sequencing. This raises the question if noise that is detected on the mRNA level propagates to fluctuations in proteins, which in turn cause phenotypic variations between individual cells. To build the connection between mRNA and protein noise as well as chromatin state and mRNA noise, multi-omics technologies need to advance in precision and scalability.

\section{Confounding effects when measuring noise}

Experimental noise
On the experimental side, cell isolation and sorting as well as culturing conditions can influence the level of heterogeneity in cell populations. It is therefore difficult to disentangle intrinsic contribution to noise from extrinsic sources due to cells committing to alternative fates (structured heterogeneity).  \\

Technical noise
As discussed above, a variety of measures exist to quantify variability in mRNA and protein abundance for populations of single cells. Due to low starting amounts in sequencing based technologies, technical noise is a major contributor to overall variation in the data. This effect is pronounced for lowly expressed genes making it harder to estimate correct noise parameters for these genes. Without multiple replicates per condition, robust assessment and testing of changes in variability is impossible.\\

\section{Measures to quantify noise}

 The distribution of pairwise distances between cells has therefore been used to describe this property(Mohammed et al. 2017). More robust measurements are nevertheless still missing.
 
 Besides empirical estimations of noise in biological systems (e.g. CV2), the development of theoretical models to describe intrinsic and extrinsic noise advanced in recent years(Fu Pachter 2016). Nevertheless, these models have not been extended to learn transcription parameters (burst size and burst frequency) or to incorporate cellular consequences of noise (e.g. cell fate decisions).

\section{Experimental validation and manipulation of noise}

Classically, unicellular systems were employed to study noise. In these systems, genetic alterations allowed the modulation of transcriptional and translational variability(Raser Oâ€™Shea 2005; Ozbudak et al. 2002; Hornung et al. 2012). Specifically, changing promoter architecture shows strong alterations of expression noise(Jones et al. 2014; Sharon et al. 2014). Furthermore, different ways of general and targeted perturbation of transcriptional noise in cell populations has been discussed in Dueck et al.(Dueck et al. 2016). So far, perturbation of biological noise in higher organisms has not been experimentally described yet.


\section{Future approaches for Bayesian models in scRNA-Seq data}

One solution for this includes error propagation approaches used in Bayesian statistics(Vallejos et al. 2016). Furthermore, these single gene measures are not optimal to quantify the global noise status of a whole population of cells.

Multi-view learning\\
Variational approaches\\
Incoporation of translational dynamics into generative models

