%!TEX root = ../main.tex
%******************************
%	 Introduction 
%*****************************

\chapter{Introduction}  

\graphicspath{{"../../Dropbox (Cambridge  University)/Figures_for_thesis/Introduction/"}}

\begin{Abstract}
The intrinsic stochasticity of biochemical reactions introduces phenotypic heterogeneity in seemingly homogeneous populations of cells. This phenomenon has been widely studied in prokaryotic and eukaryotic systems and the functional role of phenotypic variation in development, health and disease is subject of ongoing research. Biological noise, a term to describe molecular variability between individual cells, arises from different sources in homogeneous cell populations. Intrinsic noise summarizes stochastic differences in transcription and translation between individual genes \citep{Elowitz2002, Raser2004, Sanchez2013}. Extrinsic noise on the other hand arises when cells reside in different cellular states (e.g.~cell cycle, cell-to-cell signalling and metabolism)  \citep{Zopf2013, Iwamoto2016, Kiviet2014}. Recent technological advances allow the in depth analysis of biological noise in cell populations. Imaging methodologies \citep{Moffitt2016a} and single-cell “omics” techniques \citep{Bock2016} permit the quantification of thousands of mRNA species, the genomic sequence, its epigenetic modification, and selected sets of proteins per cell. Moreover, the development of multi-omics technologies introduced the possibility to link cell-to-cell variation between multiple regulatory layers across individual cells \citep{Macaulay2017}. With the emergence of \gls{scRNA-Seq} technologies, new computational strategies to quantify noise were introduced \cite{Brennecke2013, Vallejos2015, Kolodziejczyk2015cell, Buettner2015, Fan2015a, Richard2016}. Applying high-throughput scRNA-Seq to mammalian systems characterised the functional role of biological noise in healthy as well as diseased contexts. Studies of recent years described changes in noise levels at different stages during embryonic development which hints at stochastic contributions to early cell fate decisions \citep{Goolam2016, Mohammed2017, Ohnishi2014}. Phenotypic variation in immune cells, possibly derived by transcriptional noise, increases cellular plasticity and facilitates the population response to pathogens \citep{Shalek2014, Kellogg2015a}. Conversely, genetic and non-genetic heterogeneity within cell populations was described as driver for cancer development \citep{Marusyk2012} and noise increases with age \citep{Martinez-jimenez2017, Enge2017}. Here, I introduce noise as an inherent feature of biological system and discuss its positive and negative consequences in cell populations. Furthermore, I outline recent developments of single-cell sequencing and imaging technologies and comment on robust ways of quantifying biological noise. Finally, I will summarise Bayesian inference as a powerful statistical framework to model transcriptional noise from \gls{scRNA-Seq} data.
\end{Abstract}

\newpage

% Include different main sections of the introduction
\input{Introduction/Intro_files/biological_noise}
\newpage
\input{Introduction/Intro_files/sources}
\newpage
\input{Introduction/Intro_files/quantification}
\newpage
\input{Introduction/Intro_files/Other_applications}
\newpage
\input{Introduction/Intro_files/bayesian}
\newpage
\input{Introduction/Intro_files/outline}
