%!TEX root = ../chapter1.tex
%******************************
%	 Discussion 
%*****************************

\section{Discussion}

%In this chapter, we explored the effect of ageing on transcriptional programmes during immune activation. Early immune activation induces a transcriptional switch from stochastic to regulated gene expression coupled with a reduction in transcriptional variability. These dynamics and, more importantly, a set of immune-related response genes are conserved during evolution. While ageing only shows subtle effects on the overall transcriptomic profiles of individual cells, we observe a strong increase in expression variability in the core set of immune response genes during ageing. Therefore, transcriptional variability is a largely unexplored factor of organismal ageing.

How cell-type-specific gene expression programs change during organismal lifespan has long been debated \citep{Bahar2006, Warren2007} but until the beginning of the study, few studies in mammals have quantified the cell-to-cell transcriptome-wide differences that accumulate during ageing \citep{Kowalczyk2015}. Here, we systematically explored the effect of ageing on the dynamic activation program of primary naive CD4\plus{} T cells. We analysed two sub-species of mice which represents a powerful strategy to identify evolutionarily conserved gene expression programmes \citep{Shay2013}. In contrast to humans, mice were housed in specific pathogen-free facilities which reduces transcriptional changes due to pathogen-induced immune activation \citep{Beura2016}. In this chapter, we therefore profiled the intrinsit effect of ageing on transcriptional regulation in CD4\plus{} T cells.\\

By activating naive CD4\plus{} T cells and quantifying the transcriptional responses of hundreds of single-cells using scRNA-Seq, we confirmed that translation processes and immune response genes are rapidly up-regulated \citep{Neme2016, Asmal2003,Turner2014, Glass2010, Gerondakis2010, Croft2009}. More interestingly, we discovered that transcriptional variability is reduced across thousands of transcripts that otherwise remain stable in mean expression levels. This indicates that immune activation rapidly reduces transcriptional heterogeneity across the population of CD4\plus{} T cells to up-regulate a specific response programme similarly in each individual cell. A similar programme has been identified in iPS reprogramming where an early phase is characterized by probabilistic events while later on, the transcription of Sox2 induces a  more deterministic phase \citep{Buganim2012}. Previous studies assayed heterogeneity in immune responses by profiling individual cytokines such as interleukin 2 and interferon $\beta$ in immune cells. Early responding cells support the activation of surrounding cells by paracrine signalling \citep{Fuhrmann2016, Shalek2014}. In contrast, by profiling thousands of genes, our approach identifies the global collapse of variability as a key event in immune activation.\\

Comparison of gene expression levels across species have been used as a means to identify transcription under strong selection in tissues \citep{Brawand2011, Sudmant2015, Romero2012, Barbosa-Morais2012, Perry2012}, including bulk CD4\plus{} T cells from young mice and humans during immune stimulation \citep{Shay2013}. By profiling two sub-species of mice, we identified a common set of activation genes, including well-characterized immune response genes such as \textit{Il2ra}, that are similarly up-regulated across the two species. Furthermore, using scRNA-Seq allowed us to determine the number of cells that express a certain genes. With this, we newly revealed that immune stimulation results in the vast majority of cells within each species to up-regulate the set of evolutionarily conserved genes. In contrast, we discovered that genes whose mean expression was up-regulated in a species-specific manner were often activated in only a small fraction of cells, suggesting weaker selection. Indeed, species-specific up-regulated genes showed no functional enrichment. This discovery suggests a novel defining feature of functional target genes: coherent transcriptional up-regulation across a population of cells. \\

Many attempts have been made to identify transcriptional signatures associated with ageing \citep{DeMagalhaes2009, Magalhaes2009, Chen2013, Kowalczyk2015}. On a genome-wide basis, we observed that ageing has minimal effects on mean expression levels in unstimulated and stimulated CD4\plus{} T cells. However, in the core set of activated genes, in both species and in distinct CD\plus{} T cell subsets we found a markedly more heterogeneous transcriptional response to stimulation in older mice. This increased heterogeneity was driven by ageing associated differences in the reduced fraction of cells across the population that express these response genes. Instead of detecting structured heterogeneity caused by some cells not responding to the stimulus, we observed that all cells from old animals responded but in contrast to young cells, failed to homogeneously up-regulate the response programme. High numbers of CD4\plus{} T cells are needed to combat infection and cancer. The discovery that CD4\plus{} T cells from aged mice are unable to up-regulate a core activation program robustly may in part explain the decrease of immune function observed in aged mammals \citep{Goronzy2013, Nikolich-Zugich2018}. More generally, in the context of the current understanding of transcriptional dysregulation and chromatin destabilization during ageing \citep{Booth2016}, increased cell-to-cell transcriptional variability is a major, and largely unexplored, intrinsic factor.\\

Following this study, Quakes paper (pro), Liu (con)\\


Possible reasons: Desch^enes and Cheung

Furthermore, Cheung \emph{et al.} profiled a variety of epigenetic marks in different subsets peripheral blood mononuclear cells (PBMCs) in young (< 25 years) and old (> 65 years) humans on a single-cell level \citep{Cheung2018}. Analysis of 40 chromatin marks in 20 cell types revealed a separation between young and old individuals and an enrichment in most chromatin marks during ageing. Furthermore, they found an increase in cell-to-cell variability for the majority of chromatin marks in aged individuals. The authors proposed a possible role of polycomb-repressive complexes (PRC) for increasing epigenetic variation and showed that PRC-mediated H3K27me3 deposition explains the increase in transcriptional variability that we reported in this chapter \citep{Cheung2018}.

\todo{Discuss Stephen Quakes paper and epiCytof and expand on other relevant papers.}

\todo{Discuss tissue specificity.}

\todo{discuss reduced activation dynamics in CAST}

\todo{As discussed above, ... discuss mean expression confounder}





