%!TEX root = ../chapter1.tex
%******************************
%	 Discussion 
%*****************************

\section{Discussion}

How cell-type-specific gene expression programs change during organismal lifespan has long been debated \citep{Bahar2006, Warren2007} but few studies in mammals have quantified the cell-to-cell transcriptome-wide differences that accumulate during ageing \citep{Kowalczyk2015}. Here, we systematically explored the effect of ageing on the dynamic activation program of primary naive CD4\plus{} T cells in two sub-species of mice as a powerful, well-characterized, and versatile model system \citep{Shay2013}. With this system, we could discriminate the intrinsic effects of ageing from changes due to pathogen-induced immune activation by using mice housed in specific pathogen-free barrier facilities for experiments \citep{Beura2016}.\\

By activating naive CD4\plus{} T cells and quantifying the transcriptional responses of hundreds of single-cells using scRNA-Seq, we confirmed that translation processes and immune response genes are rapidly up-regulated \citep{Neme2016, Asmal2003}. We newly discovered that cell-to-cell variability is simultaneously reduced across thousands of transcripts that do not show changes in gene expression levels. In other words, immune activation rapidly reduces transcriptional heterogeneity across the otherwise-diverse population of CD4\plus{} T cells, revealing a regulatory strategy identical to how iPS reprogramming dynamically restructures transcriptional programs \citep{Buganim2012}. \\

Comparison of gene expression levels across species have been used as a means to identify transcription under strong selection in tissues \citep{Brawand2011, Sudmant2015, Romero2012, Barbosa-Morais2012, Perry2012}, including bulk CD4\plus{} T cells from young mice and humans during immune stimulation \citep{Shay2013}. As expected, we identified a common set of activation genes, including \textit{Il2ra}, that are up-regulated across sub-species; in addition, our scRNA-Seq analyses newly revealed that immune stimulation results in the vast majority of cells within each species up-regulating these genes. In contrast, we discovered that genes whose mean expression was up-regulated in a species-specific manner were often activated in only a small fraction of cells, suggesting weaker selection. Indeed, species-specific up-regulated genes showed no functional enrichment. This discovery suggests a novel defining feature of functional target genes: coherent transcriptional up-regulation across a population of cells. \\

Many attempts have been made to identify transcriptional signatures associated with ageing \citep{Magalhaes2009, Chen2013, Kowalczyk2015}. On a genome-wide basis, we observed that ageing has minimal effects on mean expression levels in unstimulated and stimulated CD4\plus{} T cells. However, in the core set of activated genes, in both species and in distinct CD\plus{} T cell subsets we found a markedly more heterogeneous transcriptional response to stimulation in older mice. Indeed, this increased heterogeneity was driven by ageing associated differences in the reduced fraction of cells across the population that fully responded to stimulation. High numbers of CD4\plus{} T cells are needed to combat infection and cancer. The discovery that CD4\plus{} T cells from aged mice are unable to up-regulate a core activation program robustly may in part explain the decrease of immune function observed in aged mammals \citep{Goronzy2013}. More generally, in the context of the current understanding of transcriptional dysregulation and chromatin destabilization during ageing \citep{Booth2016}, increased cell-to-cell transcriptional variability is a major, and largely unexplored, intrinsic factor.\\

\todo{Discuss Stephen Quakes paper and epiCytof and expand on other relevant papers.}

\todo{Discuss tissue specificity.}

\todo{discuss reduced activation dynamics in CAST}





