%!TEX root = ../chapter1.tex
%******************************
%	 Discussion 
%*****************************

\section{Discussion}

How cell-type-specific gene expression programmes change during organismal lifespan has long been debated \citep{Bahar2006, Warren2007} but until the beginning of this project, few studies in mammals have quantified the cell-to-cell transcriptome-wide differences that accumulate during ageing \citep{Kowalczyk2015}. Here, we systematically explored the effect of ageing on the dynamic activation programme of primary naive CD4\plus{} T cells. We analysed two sub-species of mice, which represents a powerful strategy to identify evolutionarily conserved gene expression programmes \citep{Shay2013}. In contrast to humans, mice were housed in specific pathogen-free facilities that reduces transcriptional changes due to pathogen-induced immune activation \citep{Beura2016}. In this chapter, we therefore profiled the intrinsic effect of ageing on transcriptional regulation in CD4\plus{} T cells.\\

By activating naive CD4\plus{} T cells and quantifying the transcriptional responses of hundreds of single-cells using scRNA-Seq, we confirmed that translation processes and immune response genes are rapidly up-regulated \citep{Asmal2003, Neme2016, Turner2014, Glass2010, Gerondakis2010, Croft2009}. More interestingly, we discovered that transcriptional variability is reduced across thousands of transcripts that otherwise remain stable in mean expression levels. This indicates that immune activation rapidly reduces transcriptional heterogeneity across the population of CD4\plus{} T cells to up-regulate a specific response programme similarly in each individual cell. A similar programme has been identified in iPSC reprogramming where an early phase is characterised by probabilistic events while, later, the transcription of \textit{Sox2} induces a  more deterministic phase \citep{Buganim2012}. Previous studies assayed heterogeneity in immune responses by profiling individual cytokines such as Il2 and Ifn\textbeta{} in immune cells. Early responding cells support the activation of surrounding cells by paracrine signalling \citep{Fuhrmann2016, Shalek2014}. In contrast, by profiling thousands of genes, our approach identifies the global collapse of variability as a key event in immune activation.\\

Comparison of gene expression levels across species have been used as a means to identify transcription under strong selection in tissues \citep{Sudmant2015, Brawand2011, Romero2012, Barbosa-Morais2012, Perry2012}, including bulk CD4\plus{} T cells from young mice and humans during immune stimulation \citep{Shay2013}. By profiling two sub-species of mice, we identified a common set of activation genes, including well-characterised immune response genes such as \textit{Il2ra}, that are similarly up-regulated across the two species. Furthermore, using scRNA-Seq allowed us to determine the number of cells that express a certain gene. With this, we newly revealed that immune stimulation results in the vast majority of cells within each species up-regulating the set of evolutionarily conserved genes. In contrast, we discovered that genes whose mean expression was up-regulated in a species-specific manner were often activated in only a small fraction of cells, suggesting weaker selection. Indeed, species-specific up-regulated genes showed no functional enrichment. This discovery suggests a novel defining feature of functional target genes: coherent transcriptional up-regulation across a population of cells. \\

Many attempts have been made to identify transcriptional signatures associated with ageing \citep{DeMagalhaes2009, Magalhaes2009, Chen2013, Kowalczyk2015}. On a genome-wide basis, we observed that ageing has minimal effects on mean expression levels in unstimulated and stimulated CD4\plus{} T cells. However, in the core set of activated genes, in both species and in distinct CD\plus{} T cell subsets, we found a markedly more heterogeneous transcriptional response to stimulation in older mice. This increased heterogeneity was driven by ageing associated differences in the (reduced) fraction of cells across the population that express these response genes. Instead of detecting structured heterogeneity, characterised by some cells not responding to the stimulus, we observed that all cells from old animals responded, but in contrast to young cells, failed to homogeneously up-regulate the response programme. \\

High numbers of CD4\plus{} T cells are needed to combat infection and cancer. The discovery that CD4\plus{} T cells from aged mice are unable to robustly up-regulate a core activation programme may in part explain the decrease of immune function observed in aged mammals \citep{Goronzy2013, Nikolich-Zugich2018}. More generally, in the context of the current understanding of transcriptional dysregulation and chromatin destabilisation during ageing \citep{Booth2016}, increased cell-to-cell transcriptional variability is a major, and largely unexplored, intrinsic factor.\\

Following the publication of this study, several mechanisms for the increase in transcriptional variability during ageing have been proposed. In one study, the increase in transcriptional variability during ageing was confirmed in human pancreatic \textbeta{}-cells. A possible mechanism for this increase in variability is so called "fate drift" of \textbeta{}-cells to resemble \textalpha{}-cells. During ageing, \textbeta{}-cells that are defined by their expression of the hormone \emph{insulin} increase expression of the hormone \emph{glucagon}, the characteristic hormone of \textalpha{}-cells. This atypical hormone expression can result in increased transcriptional noise during ageing in the pancreas \citep{Enge2017}. Another study by Desch\^{e}nes and Chabot, 2017 proposed that the stochastic shortening of telomeres during ageing introduces variation in a process termed \gls{TPE-OLD}. TPE-OLD regulates the expression of genes 10 Mb into the chromosome and its variation can lead increased transcriptional heterogeneity during ageing \cite{Deschenes2017}. Further, Cheung \emph{et al.}, 2018 profiled a variety of epigenetic marks in different subsets \glspl{PBMC} in young (< 25 years) and old (> 65 years) humans at single-cell resolution \citep{Cheung2018}. Analysis of 40 chromatin marks in 20 cell types revealed a separation between young and old individuals and an enrichment in most chromatin marks during ageing. Furthermore, they found an increase in cell-to-cell variability for the majority of chromatin marks in aged individuals. The authors proposed a possible role for \gls{PRC} in increasing epigenetic variation and showed that PRC-mediated H3K27me3 deposition explains the increase in transcriptional variability that we reported in this chapter \citep{Cheung2018}.\\

While an increase in transcriptional noise has been shown to be associated with tissue ageing in pancreas and the immune system, a more complete view of whole-organism tissue ageing is missing. One study that begins to address this profiled changes in transcriptional noise in multiple cell types in young and old mice \citep{Angelidis2018}. Not only did they confirm the increase in transcriptional noise during ageing in CD4\plus{} T cells but also observed this shift in a variety of cell types associated with the lung (e.g. NK cells, macrophages, dendritic cells, endothelial cells, smooth muscle cells and neutrophils) \citep{Angelidis2018}. This analysis validates increased transcriptional noise as a major hallmark of ageing. \\

The major drawback in this chapter was the inability to profile all immune response genes for changes in variability due to the dependency of the over-dispersion parameter on the mean expression parameters (see \textbf{Section \ref{sec0:BASiCS}}). The simple approach to only profile genes with stable mean expression levels during immune activation excluded all immune-associated genes from analysis. These are generally the genes that define T cell phenotypes and functionality. In the next chapter, I will therefore describe an extension of the BASiCS framework to include genes that display changes in mean expression by regressing out this mean-variability dependency.





