%!TEX root = ../chapter1.tex
%******************************
%	 Introduction 
%*****************************

\section{Introduction} 

Ageing is characterized by the progressive decline of physiological and cellular functions \citep{Lopez-Otin2013, Booth2016}. It can have a complex and tissue-specific impact on gene expression levels \citep{Zahn2007}, as seen by microarray expression analyses of collections of mouse CD4$^+$ and CD8$^+$ T cells \citep{Mirza2011}, rat hepatocytes \citep{Tollet-Egnell2000}, mouse and human brain \citep{Lu2004, Lee2000}, human muscle \citep{Welle2003, Zahn2006}, human kidney \citep{Rodwell2004}, human retina \citep{Yoshida2002}, and different species of Drosophila and Caenorhabditis \citep{Mccarroll2004}. For instance, aging affects distinct functional pathways, even in closely related CD4$^+$ and CD8$^+$ T cells \citep{Mirza2011}. \\

Approaches that analyse the expression of sets of genes on a single-cell basis have more recently suggested that ageing may also alter the cell-to-cell variability of gene expression. Analysis of fifteen genes in terminally differentiated cardiomyocytes suggested that aging can lead to increased cell-to-cell transcriptional variability \citep{Bahar2006}. In contrast, single-cell analysis of the transcription of six genes in four different hematopoietic stem cell types showed few cell-to-cell changes between old and young animals, leading to the suggestion that transcriptional instability may not be a universal attribute of ageing \citep{Warren2007}. Whether cell-to-cell gene expression variability increases during ageing on a genome-wide basis, particularly for dynamic activation programs, remains largely unexplored.\\

Single-cell RNA sequencing can now allow the quantification of transcriptional variability in thousands of genes simultaneously. For example, Kowalczyk \textit{et al.} performed a high-resolution scRNA-seq analysis of hematopoietic stem cells in young and old mice. Here, cell cycle is the primary driver for cell-to-cell variability in gene expression, and aging decreases the entry of long-term hematopoietic stem cells into G1 phase in a cell-type-specific manner \citep{Kowalczyk2015}.\\ 

Naive CD4$^+$ T cells are an excellent model system to evaluate the impact of ageing on gene expression levels and cell-to-cell transcriptional variability. They are readily isolated as single, phenotypically homogeneous cells when purified from young and aged spleens and can be easily stimulated into a physiologically-relevant activated transcriptional state in vitro. Naive CD4$^+$ T cells are maintained in a quiescent state, but have the ability to respond to antigen stimulation with proliferation and effector differentiation, which is essential for life-long maintenance of adaptive immune function against infection and cancer \citep{Swain2012, Kim2014a}. \\

A conserved set of response genes has been identified by comparison of bulk gene expression between human and mouse CD4$^+$ T cells after immune activation \citep{Shay2013}. Indeed, as a strategy, comparison of gene expression levels in matched tissues from different mammalian species is a powerful tool for revealing conserved cell-type-specific regulatory programs \citep{Sudmant2015, Finseth2014, Brawand2011, Flajnik2009}. It is not known whether conservation of gene expression levels is also reflected in cell-to-cell variability.\\

Here, we dissect the activation dynamics of naive CD4$^+$ T cells at the single cell level during ageing in two sub-species of mice. In young and old animals, immune activation causes the up-regulation of hundreds of target genes, coupled to a strong decrease in cell-to-cell transcriptional variability across all genes. Ageing had only modest and species-specific effects on gene expression levels, but substantially increased cell-to-cell transcriptional variability in naive cells and effector memory cells upon activation.
